python/ray/tests/aws/test_autoscaler_aws.py

```diff
@@ -119,10 +119,16 @@ def test_create_sg_different_vpc_same_rules(iam_client_stub, ec2_client_stub):
 
     # expect the bootstrapped config to show different head and worker security
     # groups residing on different subnets
-    assert config["head_node"]["SecurityGroupIds"] == [DEFAULT_SG["GroupId"]]
-    assert config["head_node"]["SubnetIds"] == [DEFAULT_SUBNET["SubnetId"]]
-    assert config["worker_nodes"]["SecurityGroupIds"] == [AUX_SG["GroupId"]]
-    assert config["worker_nodes"]["SubnetIds"] == [AUX_SUBNET["SubnetId"]]
+    for node_type_key, node_type in config["available_node_types"].items():
+        node_config = node_type["node_config"]
+        security_group_ids = node_config["SecurityGroupIds"]
+        subnet_ids = node_config["SubnetIds"]
+        if node_type_key == config["head_node_type"]:
+            assert security_group_ids == [DEFAULT_SG["GroupId"]]
+            assert subnet_ids == [DEFAULT_SUBNET["SubnetId"]]
+        else:
+            assert security_group_ids == [AUX_SG["GroupId"]]
```

Comment:
Is this "Custom SG"? Why there are different SGs based on node type?

Reply:
Yeah, it's an extra/auxiliary SG.
There was support added in this [PR](https://github.com/ray-project/ray/pull/8374) a year ago for launching head and workers in separate VPCs -- that requires creating different security groups for head and workers. This test checks that this works. The changes here just port the old test into the new format.
With the changes in this PR, one should be able different node types in different VPCs -- I think few users need this functionality, so it's not worth thinking too much about this.